packages/neuron-wallet/src/services/tx/transaction-service.ts

```diff
@@ -485,29 +482,45 @@ export class TransactionsService {
   }
 
   public static async get(hash: string): Promise<Transaction | undefined> {
-    const tx = await getConnection()
+    const txInDB = await getConnection()
       .getRepository(TransactionEntity)
       .createQueryBuilder('transaction')
       .where('transaction.hash is :hash', { hash })
-      .leftJoinAndSelect('transaction.inputs', 'input')
-      .orderBy({
-        'input.id': 'ASC',
-      })
       .getOne()
-    const txOutputs = await getConnection()
-      .getRepository(OutputEntity)
-      .createQueryBuilder()
-      .where({
-        outPointTxHash: hash,
-      })
-      .getMany()
-
-    if (!tx) {
+    if (!txInDB) {
       return undefined
     }
-
-    tx.outputs = txOutputs
-    return tx.toModel()
+    const url: string = NetworksService.getInstance().getCurrent().remote
+    const rpcService = new RpcService(url)
+    const txWithStatus = await rpcService.getTransaction(hash)
+    if (!txWithStatus?.transaction) {
+      return undefined
+    }
+    const tx = txInDB.toModel()
+    const inputTxHashes = txWithStatus.transaction.inputs.map(v => v.previousOutput?.txHash).filter((v): v is string => !!v)
+    const ckb = new CKB(url)
+    const inputTxs = await ckb.rpc
+      .createBatchRequest<'getTransaction', string[], CKBComponents.TransactionWithStatus[]>(
+        inputTxHashes.map(v => ['getTransaction', v])
+      )
+      .exec()
+    const inputTxMap = new Map<string, CKBComponents.Transaction>()
+    inputTxs.forEach((v, idx) => {
+      inputTxMap.set(inputTxHashes[idx], v.transaction)
+    })
+    txWithStatus.transaction.inputs.forEach(v => {
```

Comment:
The operation performed by this for loop leaves me a bit confused, so it might lack comments, or it should be extracted into a separate function.